### PR TITLE
Changes to support more distros in Sysbox. 

### PR DESCRIPTION
**Pull request by ctalledo**

### CODE REVIEW

File: idShiftUtils/go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/nestybox/sysbox/idShiftUtils
+
+go 1.14
```
Comment (rodnymolina) [inline idShiftUtils/go.mod]: This reminds me that we should do this in all modules: s / 1.1? / 1.16

We 'own' the build-infra so it's up to us to define the compilation requirements.

File: utils/linux.go [hunk 2]
```diff
@@ -180,6 +183,32 @@ func KernelCurrentVersionCmp(k1Major, k1Minor int) (int, error) {
 	return -1, nil
 }
 
+// Parses the kernel release string (obtained from GetKernelRelease()) and returns
```
Comment (rodnymolina) [inline utils/linux.go:186]: Don't think this function is needed now that we have `KernelCurrentVersionCmp()` above.

File: utils/linux.go [hunk 3]
```diff
@@ -203,3 +232,50 @@ func GetLinuxHeaderPath(distro string) (string, error) {
...
+	if major < 5 {
+		return false, nil
+	} else if major == 5 && minor < 12 {
+		return false, nil
+	} else {
+		return true, nil
+	}
+}
+
+// KernelModSupported returns nil if the given module is loaded in the kernel.
```
Comment (rodnymolina) [inline utils/linux.go:260]: Good idea to have something like this.

File: utils/linux.go [hunk 4]
```diff
@@ -203,3 +232,50 @@ func GetLinuxHeaderPath(distro string) (string, error) {
...
+func KernelModSupported(mod string) (bool, error) {
+
+	// Load the module
+	exec.Command("modprobe", mod).Run()
+
+	// Check if the module is in the kernel
```
Comment (rodnymolina) [inline utils/linux.go:266]: Why isn't it enough to check `modprobe` return value to save us the cycles of reading `/proc/modules` (usually a 100+ lines file)?

**Review by rodnymolina — APPROVED** [review]
Looks good. Have a few comments / suggestions.

File: idShiftUtils/go.mod (hunk 1 above)
Comment (ctalledo) [reply idShiftUtils/go.mod]: Agreed. Done for idShiftUtils.

File: utils/linux.go (hunk 2 above)
Comment (ctalledo) [reply utils/linux.go:186]: I still think the function is useful as a way to parse kernel version numbers (i.e., one may want to parse them for reasons other than comparing them against the current kernel version).

File: utils/linux.go (hunk 4 above)
Comment (ctalledo) [reply utils/linux.go:266]: Good point. At the time, I wanted to remove the dependency on the `modprobe` command and get the info directly from the kernel (which explains why we don't check the output of modprobe in the code). I guess we could rely solely on modprobe, but it's not clear to me if its included in all distros (though I suspect it is).

**Pull request merged**